signalclient.go

```diff
@@ -348,18 +350,17 @@ func (c *SignalClient) handleResponse(res *livekit.SignalResponse) {
 func (c *SignalClient) readWorker() {
 	defer func() {
 		c.isStarted.Store(false)
-		if c.OnClose != nil {
-			c.OnClose()
```

Comment:
This means `OnClose` will only be called when `Close` is called instead of connection is interrupted, and the `Engine` relies this callback to handle signal disconnection and resume.

Reply:
Thank you @cnderrauber . Will take a look. Was hitting a data race in the tests with the `OnClose()` in the `defer`. So, I serialised it :-( My bad. Will take a look at calling back on read loop exit.

Reply:
@cnderrauber Made changes in these two commits
- https://github.com/livekit/server-sdk-go/pull/351/commits/9a727bc78b3482d2a8e4177e8391f80902d19618 - moving `OnClose()` callback to `defer` on `readWorker`.
- https://github.com/livekit/server-sdk-go/pull/351/commits/0acef170f0908065abd1546e9cee319bbeb0ea19 - using a local variable to avoid race. The race reported before is something like [this](https://github.com/livekit/server-sdk-go/actions/runs/6926214815/job/18838032115). The test was calling simulate scenario reconnect which called signal client close. And the read on `readerCloserCh` in `Close()` was being accessed in both `Close()` and `Start()` (where it is changed to set up for next read worker). Getting around it by using a local variable in `Close()` and read on that.